caveclient/skeletonservice.py

```diff
@@ -240,6 +247,102 @@ def build_endpoint(
         url = self._endpoints[endpoint].format_map(endpoint_mapping)
         return url
 
+    def _build_bulk_endpoint(
+        self,
+        root_ids: List,
+        datastack_name: str,
+        skeleton_version: int,
+        output_format: str,
+        generate_missing_sks: bool,
+    ):
+        """
+        Building the URL in a separate function facilitates testing
+        """
+        if datastack_name is None:
+            datastack_name = self._datastack_name
+        assert datastack_name is not None
+
+        endpoint_mapping = self.default_url_mapping
+        endpoint_mapping["datastack_name"] = datastack_name
+        endpoint_mapping["root_ids"] = ",".join([str(v) for v in root_ids])
+        endpoint_mapping["output_format"] = output_format
+        endpoint_mapping["gen_missing_sks"] = generate_missing_sks
+
+        if not skeleton_version:
+            endpoint = "get_bulk_skeletons_via_rids"
+        else:
+            endpoint_mapping["skeleton_version"] = skeleton_version
+            endpoint = "get_bulk_skeletons_via_skvn_rids"
+
+        url = self._endpoints[endpoint].format_map(endpoint_mapping)
+        return url
+
+    def _build_bulk_async_endpoint(
+        self,
+        root_ids: List,
+        datastack_name: str,
+        skeleton_version: int,
+    ):
+        """
+        Building the URL in a separate function facilitates testing
+        """
+        if datastack_name is None:
+            datastack_name = self._datastack_name
+        assert datastack_name is not None
+
+        endpoint_mapping = self.default_url_mapping
+        endpoint_mapping["datastack_name"] = datastack_name
+        endpoint_mapping["root_ids"] = ",".join([str(v) for v in root_ids])
+
+        if not skeleton_version:
+            endpoint = "gen_bulk_skeletons_via_rids"
+        else:
+            endpoint_mapping["skeleton_version"] = skeleton_version
+            endpoint = "gen_bulk_skeletons_via_skvn_rids"
+
+        url = self._endpoints[endpoint].format_map(endpoint_mapping)
+        return url
+
+    def get_cache_contents(
```

Comment:
This needs a server versioning check so that users don't get 404 errors when querying servers that have not upgraded to include this endpoint

Reply:
Ah. Thanks!